baseplate/lib/tracing.py

```diff
@@ -42,3 +44,72 @@ def should_sample(
 
     def get_description(self) -> str:
         return f"RateLimited(fixed rate sampling {self.rps})"
+
+
+# Greenlet tracing utils
+__Greenlet = gevent.Greenlet
+__IMap = gevent.pool.IMap
+__IMapUnordered = gevent.pool.IMapUnordered
+
+
+class Runnable(Protocol):
+    @property
+    def trace_context(self) -> Context: ...
+
+    run: Callable
+
+
+class TracingMixin:
+    def __init__(self: Runnable, *args, **kwargs) -> None:  # type: ignore
+        self.trace_context = context.get_current()
```

Comment:
nit: if there aren't naming restrictions, maybe name it `baseplate_trace_context` to avoid potential future clashes with the gevent class?

Reply:
Good call, made the change.